File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/kotlin/jaimefere.kt
fun isPrime(number: Int): Boolean {
    if(number < 2) {
        return false
    } else {
        (2 until number).forEach { partialNumber ->
            if(number % partialNumber == 0) {
                return false
            }
        }
        return true
    }
}

fun isFibonacci(number: Int): Boolean {
    var penultimate = 0
    var ultimate = 1
    while(ultimate < number) {
        ultimate += penultimate
        penultimate = ultimate - penultimate
    }
    return (number == penultimate) || (number == ultimate)
}

fun isEven(number: Int): Boolean {
    return number % 2 == 0
}

fun getKindOf(number: Int): String {
    return "$number ${if(isPrime(number)) "es" else "no es"} primo, ${if(isFibonacci(number)) "es" else "no es"} fibonacci y ${if(isEven(number)) "es par" else "es impar"}"
}

fun main() {
    println(getKindOf(0))
    println(getKindOf(1))
    println(getKindOf(2))
    println(getKindOf(7))
}

File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/swift/jaimefere.swift
import Foundation

func isPrime(_ number: Int) -> Bool {
    if(number < 2) {
        return false
    } else {
        var result = true
        (2..<number).forEach { partialNumber in
            if(number % partialNumber == 0) {
                result = false
            }
        }
        return result
    }
}

func isFibonacci(_ number: Int) -> Bool {
    var penultimate = 0
    var ultimate = 1
    while(ultimate < number) {
        ultimate += penultimate
        penultimate = ultimate - penultimate
    }
    return number == penultimate || number == ultimate
}

func isEven(_ number: Int) -> Bool {
    return number % 2 == 0
}

func getKindOf(_ number: Int) -> String {
    return "\(number) \(isPrime(number) ? "es" : "no es") primo, \(isFibonacci(number) ? "es" : "no es") fibonacci y \(isEven(number) ? "es par" : "es impar")"
}

print(getKindOf(0))
print(getKindOf(1))
print(getKindOf(2))
print(getKindOf(7))
